### PR TITLE
Feat/update intro

### PR DESCRIPTION
**Pull request by samajammin**
# Description

Expand intro doc page

## Related issue(s)

Pulling out these changes that are ready from this WIP PR: #1046

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://github.com/privacy-scaling-explorations/maci/blob/dev/CONTRIBUTING.md) and [code of conduct](https://github.com/privacy-scaling-explorations/maci/blob/dev/CODE_OF_CONDUCT.md).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: website/versioned_docs/version-v1.x/introduction.md [hunk 1]
```diff
@@ -58,7 +71,23 @@ MACI was originally developed by:
 - [Koh Wei Jie](https://kohweijie.com)
 - [Samuel Gosling](https://twitter.com/xGozzy)
 
-Currently, it is being maintained by the new MACI team, which will continue to work on improving the protocol and its documentation.
+**[MACI v1.1.1](/blog/maci-v1-1-1-release) was primarily developed by:**
+
+- [ctrlc03](https://github.com/ctrlc03)
```
Comment (ctrlc03) [inline website/versioned_docs/version-v1.x/introduction.md:76]: I would add Cory here as well, https://github.com/privacy-scaling-explorations/maci/releases/tag/v1.1.0 && https://github.com/privacy-scaling-explorations/maci/releases/tag/v1.1.1- 
I personally only fixed audit issues so can even remove me from there

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thank you, just left a comment! Also please ensure you use conventional commits all the times, CI is complaining!

File: website/versioned_docs/version-v1.x/introduction.md (hunk 1 above)
Comment (samajammin) [inline website/versioned_docs/version-v1.x/introduction.md:76]:
```suggestion
- [corydickson](https://github.com/corydickson)
```

Comment (samajammin) [reply website/versioned_docs/version-v1.x/introduction.md]: Updated, thanks. Will do 😄

**Review by ctrlc03 — APPROVED** [review]

**Review by kittybest — APPROVED** [review]
I think there are some overlap changes as PR#1046?

**Pull request merged**